app/mixins/messages_base.js

```diff
@@ -0,0 +1,130 @@
+import Ember from "ember";
+
+export default Ember.Mixin.create({
+  body: "",
+  messages: [],
+
+  store: Ember.inject.service(),
+  subscription: Ember.inject.service(),
+  messagesUtil: Ember.inject.service("messages"),
+
+  sortProperties: ["createdAt: asc"],
+  sortedMessages: Ember.computed.sort("messages", "sortProperties"),
+
+  groupedMessages: Ember.computed("sortedMessages.[]", function() {
+    return this.groupBy(this.get("sortedMessages"), "createdDate");
+  }),
+
+  on() {
+    this.get("subscription").on("change:message", this, this.markReadAndScroll);
+  },
+
+  off() {
+    this.get("subscription").off(
+      "change:message",
+      this,
+      this.markReadAndScroll
+    );
+  },
+
+  groupBy: function(content, key) {
+    var result = [];
+    var object, value;
+
+    content.forEach(function(item) {
+      value = item.get(key);
+      object = result.findBy("value", value);
+      if (!object) {
+        object = {
+          value: value,
+          items: []
+        };
+        result.push(object);
+      }
+      return object.items.push(item);
+    });
+    return result.getEach("items");
+  },
+
+  createMessage(values) {
+    var message = this.store.createRecord("message", values);
+    message
+      .save()
+      .then(data => {
+        this.set("body", "");
+        this.get("messages").pushObject(data._internalModel);
+        this.set("displayText", "");
+      })
+      .catch(error => {
+        this.store.unloadRecord(message);
+        throw error;
+      });
+  },
+
+  prepareMessageObject: function(messageableType) {
+    Ember.$("textarea").trigger("blur");
```

Comment:
Good idea creating a mixin for messages. However we can't really have JQuery code inside a mixin, for the simple reason that it should be reusable within any component/controller. So you can't assume that there is a `textarea` in the page.

That part can be kept in the controller : )